src/azul/service/async_manifest_service.py

```diff
@@ -61,43 +76,74 @@ class AsyncManifestService(AbstractService):
     """
     Starting and checking the status of manifest generation jobs.
     """
-    step_function_helper = StepFunctionHelper()
+    helper = StepFunctionHelper()
+
+    step_function_execution_retries = 3
 
     def __init__(self, state_machine_name):
         self.state_machine_name = state_machine_name
 
-    def start_generation(self, input: JSON) -> Token:
-        execution_id = str(uuid.uuid4())
-        self.step_function_helper.start_execution(self.state_machine_name,
-                                                  execution_id,
-                                                  execution_input=input)
-        return Token(execution_id=execution_id,
-                     request_index=0,
-                     wait_time=self._get_next_wait_time(0))
+    def start_generation(self, input: JSON, execution_id: str) -> Token:
+
+        for retry_index in range(self.step_function_execution_retries):
```

Comment:
What happens if the loop ends naturally, exhausting the retries? Shouldn't that result in an error? If this is a bug, why doesn't the test catch that?